plugin/grpctrace/interceptor.go

```diff
@@ -43,9 +43,29 @@ var (
 	messageUncompressedSizeKey = kv.Key("message.uncompressed_size")
 )
 
+type messageType string
+
+// Event adds an event of the messageType to the span associated with the
+// passed context with id and size (if message is a proto message).
+func (m messageType) Event(ctx context.Context, id int, message interface{}) {
+	span := trace.SpanFromContext(ctx)
+	if p, ok := message.(proto.Message); ok {
+		span.AddEvent(ctx, "message",
+			messageTypeKey.String(string(m)),
+			messageIDKey.Int(id),
+			messageUncompressedSizeKey.Int(proto.Size(p)),
+		)
+	} else {
+		span.AddEvent(ctx, "message",
+			messageTypeKey.String(string(m)),
```

Comment:
should this be string() or should it utilize Sprint()?

Reply:
`string()` is correct here as it is converting to the underlying type of the `messageType` (`string`). There is no `String()` method.

Reply:
still confused -- is m a generic `interface{}` or is it just a handy way to pass either a `Proto` or a `string`?

Reply:
also, please consider a type switch here, as it makes the behavior a little bit more obvious.

Reply:
Hmm, I think there is confusion between `m` with `message`. `m` is a `messageType`, a new type added to ensure only two message types are allowed.

Reply:
oh, aha, yup, that would do it. having one variable named `m` and the other named `message` is incredibly confusing.